lightning/backend/local.go

```diff
@@ -949,23 +950,49 @@ loopWrite:
 			for errCnt < maxRetryTimes {
 				log.L().Debug("ingest meta", zap.Reflect("meta", meta))
 				var resp *sst.IngestResponse
-				resp, err = local.Ingest(ctx, meta, region)
-				if err != nil {
-					log.L().Warn("ingest failed", log.ShortError(err), zap.Reflect("meta", meta),
-						zap.Reflect("region", region))
-					errCnt++
-					continue
-				}
+				doIngest := true
 				failpoint.Inject("FailIngestMeta", func(val failpoint.Value) {
+					// only inject the error once
 					switch val.(string) {
 					case "notleader":
-						resp.Error.NotLeader = &errorpb.NotLeader{
-							RegionId: region.Region.Id, Leader: region.Leader}
+						if errCnt == 0 {
+							resp = &sst.IngestResponse{
+								Error: &errorpb.Error{
+									NotLeader: &errorpb.NotLeader{
+										RegionId: region.Region.Id,
+										Leader:   region.Leader,
+									},
+								},
+							}
+							errCnt++
+						}
 					case "epochnotmatch":
-						resp.Error.EpochNotMatch = &errorpb.EpochNotMatch{
-							CurrentRegions: []*metapb.Region{region.Region}}
+						if retryCnt == 0 {
+							resp = &sst.IngestResponse{
+								Error: &errorpb.Error{
+									EpochNotMatch: &errorpb.EpochNotMatch{
+										CurrentRegions: []*metapb.Region{region.Region},
+									},
+								},
+							}
+							retryCnt++
```

Comment:
miss `errCnt ++` ?

Reply:
Not really. Since `epochnotmatch` error will trigger a retry of write in the other loop, increase the local variable errCnt in the inner loop makes no sense.

Reply:
ok, so `retryCnt` is more like a bool variable.

Reply:
yes, in the current logic we only let the error fail once. Maybe in the future, we will add more tests that need to fail more times.